### DISCUSSION TITLE
New features

### DISCUSSION THREAD

**Discussion by pinmarva** [Netflix/lemur#4695] [General]
Hi Lemur team!

I would like to collaborate with the project. I am interested in the following features, but I have some doubts. And I would like to know if you will be OK with them:

- Source plugin: Digicert
- Source plugin: AWS ACM (Should I implement this feature in the current AWS plugin or create a new one?)
- Destination plugin: AWS ACM  (Should I implement this feature in the current AWS plugin or create a new one?)
- Destination plugin: AWS Parameter Store (Should I implement this feature in the current AWS plugin or create a new one?)
- Issuer plugin: AWS ACM Private CA

Best regards!

**Comment by jmcrawford45** [comment]
Hi there,

Thank you for your interest in collaborating with the Lemur project! We're always excited to welcome contributors.

Regarding the Digicert source plugin, you might want to know that we already have a stub source plugin available [here](https://github.com/Netflix/lemur/blob/9a5b62a63e0cfe867d76ac0d0e4485766b5de5c6/lemur/plugins/lemur_digicert/plugin.py#L289). You're welcome to extend it, but we'll likely want this configurable (and default to off) to avoid any unexpected behaviors / breakages for existing users.

For the AWS ACM source and destination plugins, we do agree that they would be valuable additions to the project. In fact, it's a frequently requested feature, as you can see from our open issues [here](https://github.com/Netflix/lemur/issues?q=is%3Aissue+is%3Aopen+acm).

As for whether the new functionality should be added to the existing AWS plugin or a new one, our recommendation would be to create a new plugin. This is in line with our current pattern of creating separate plugins for different AWS services, such as the differentiation between the `AwsDestinationPlugin` (IAM only) and the `S3DestinationPlugin`, despite both being AWS services. Once we've added an AWS ACM source plugin, in addition to the normal doc updates, we can add a pointer to the plugin here https://github.com/Netflix/lemur/blob/9a5b62a63e0cfe867d76ac0d0e4485766b5de5c6/lemur/plugins/lemur_aws/plugin.py#L192.

Finally, the addition of an issuer plugin for AWS ACM Private CA sounds like a great idea, and we would be happy to see it.

Looking forward to your contributions!

Best Regards,
Jared
